darwin/DarwinProcess.c

```diff
@@ -498,6 +501,10 @@ void DarwinProcess_scanThreads(DarwinProcess* dp, DarwinProcessTable* dpt) {
          ProcessTable_add(&dpt->super, tprocess);
    }
 
+   if (isProcessStuck) {
+      dp->super.state = UNINTERRUPTIBLE_WAIT;
```

Comment:
This should be pushed into the thread-specific state for the affected `tid` too.

As is, this is only for the overall process.